### PR TITLE
make build_mpt_alibi_tensor a method of MptModel so that deepspeed co…

### PR DESCRIPTION
**Pull request by sywangyi**
…uld override it to make autoTP work

# What does this PR do?

Fixes # (issue)
enable autoTP for mpt in huggingface model.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker and @younesbelkada

### CODE REVIEW

**Comment by sywangyi** [comment]
@sgugger @ArthurZucker please review. thanks

**Comment by sywangyi** [comment]
should work with https://github.com/microsoft/DeepSpeed/pull/4062

File: src/transformers/models/mpt/modeling_mpt.py [hunk 1]
```diff
@@ -413,6 +390,28 @@ def __init__(self, config: MptConfig):
...
+        base = base * (alibi_bias_max / num_heads_power_of_2)
+
+        slopes = 1.0 / torch.pow(2, base)
+        slopes = slopes.view(1, num_heads, 1, 1)
+
+        if num_heads_power_of_2 != num_heads:
+            slopes = torch.concat([slopes[1::2], slopes[::2]])[:num_heads]
+
+        alibi = alibi * slopes
+        return alibi.squeeze(0)
```
Comment (ArthurZucker) [inline src/transformers/models/mpt/modeling_mpt.py:413]:
```suggestion
    def build_mpt_alibi_tensor(self,  num_heads, sequence_length, alibi_bias_max=8, device=None):
        return  build_mpt_alibi_tensor( num_heads, sequence_length, alibi_bias_max, device)
```
If we want to follow what is done in Bloom, we should probably go with that. 
I am fine with both.

Comment (ArthurZucker) [reply src/transformers/models/mpt/modeling_mpt.py:413]: Will let @sgugger decide

**Review by ArthurZucker — COMMENTED** [review]
Thanks for the PR, I guess it makes sense since bloom was also adapted. 
Can you run `make style` to pass the CI.

File: src/transformers/models/mpt/modeling_mpt.py (hunk 1 above)
Comment (sywangyi) [reply src/transformers/models/mpt/modeling_mpt.py]: thanks. @ArthurZucker for the review. let's wait for @sgugger decision. I'm ok with both two.

Comment (sgugger) [reply src/transformers/models/mpt/modeling_mpt.py]: Let's be consistent with BLOOM if it's the same?

**Review by sgugger — APPROVED** [review]
Thanks for iterating!

**Pull request merged**